services/remotecluster/service.go

```diff
@@ -55,7 +67,7 @@ type Service struct {
 	mux              sync.RWMutex
 	active           bool
 	leaderListenerId string
-	topicListeners   map[string][]TopicListener
+	topicListeners   map[string]map[string]topicListenerEntry
```

Comment:
I needed multiple listeners per topic.  Also needed to keep a listener id per listener since Go doesn't let us store function pointers as map keys.  Nested maps instead of map of lists is to ease add/remove as you mentioned at the cost of map of maps.